Core/SymbolValidation/SymbolValidator.cs

```diff
@@ -269,24 +263,52 @@ private async Task<SymbolValidatorResult> CalculateValidity(IReadOnlyList<IFile>
                     }
                     else if (IsPublicPackage)
                     {
-                        // try to get on NuGet.org
-                        // https://www.nuget.org/api/v2/symbolpackage/Newtonsoft.Json/12.0.3 -- Will redirect
-
-#pragma warning disable CA2234 // Pass system uri objects instead of strings
-#pragma warning disable CA1308 // Normalize strings to uppercase
-                        using var response = await _httpClient.GetAsync($"https://globalcdn.nuget.org/symbol-packages/{_package.Id.ToLowerInvariant()}.{_package.Version.ToNormalizedString().ToLowerInvariant()}.snupkg", cancellationToken).ConfigureAwait(false);
-#pragma warning restore CA1308 // Normalize strings to uppercase
-#pragma warning restore CA2234 // Pass system uri objects instead of strings
+                        var sourceRepository = Repository.CreateSource(Repository.Provider.GetCoreV3(), _packageSource);
+                        var symbolPackagePublishSourceUri = (await sourceRepository.GetResourceAsync<SymbolPackageUpdateResourceV3>(cancellationToken).ConfigureAwait(false))?.SourceUri;
 
-                        if (response.IsSuccessStatusCode) // we'll get a 404 if none
+                        if (symbolPackagePublishSourceUri is not null)
                         {
+                            var variants = new[]
+                            {
+                                async () => await ReadPublicPackageSymbolFile(_package.Id.ToLowerInvariant(), _package.Version.ToNormalizedString().ToLowerInvariant()), // nuget.org
+                                async () => await ReadPublicPackageSymbolFile(_package.Id, _package.Version.OriginalVersion), // myget.org
+                            };
+
+                            foreach (var variant in variants)
```

Comment:
Can we be more intelligent here so we're not blindly trying requests?

Reply:
Would be nice, but I couldn't figure it out in the short time I spend on this feature, so that I could check that our nightly build packages were properly configured in regards to symbol support.

Reply:
Feel free to directly commit changes to the PR until it reaches the quality necessary to be merged, if you want to (probably quicker).